File: WorldManager/src/main/java/com/n9mtq4/exmcl/tab/worldmanager/InitWorldManagerTab.java
package com.n9mtq4.exmcl.tab.worldmanager;

import com.n9mtq4.exmcl.tab.worldmanager.ui.WorldManagerTab;
import com.n9mtq4.logwindow.BaseConsole;
import com.n9mtq4.logwindow.events.ObjectEvent;
import com.n9mtq4.logwindow.listener.ObjectListener;
import net.minecraft.launcher.ui.tabs.LauncherTabPanel;

import javax.swing.SwingUtilities;

/**
 * Created by will on 8/31/15 at 11:39 PM.
 */
public final class InitWorldManagerTab implements ObjectListener {
	
	@Override
	public final void objectReceived(final ObjectEvent e, final BaseConsole baseConsole) {
		
		if (e.getMessage().equalsIgnoreCase("tabsafe") && e.getObject() instanceof LauncherTabPanel) {
			
			SwingUtilities.invokeLater(new Runnable() {
				@Override
				public void run() {
					WorldManagerTab worldManagerTab = new WorldManagerTab((LauncherTabPanel) e.getObject(), baseConsole);
					baseConsole.push(new Object[]{"Worlds", worldManagerTab}, "addtab");
				}
			});
			
			baseConsole.disableListenerAttribute(this);
			
		}
		
	}
	
}

File: WorldManager/src/main/kotlin/com/n9mtq4/exmcl/tab/worldmanager/InitWorldManagerTab.kt
package com.n9mtq4.exmcl.tab.worldmanager

import com.n9mtq4.exmcl.tab.worldmanager.ui.WorldManagerTab
import com.n9mtq4.logwindow.BaseConsole
import com.n9mtq4.logwindow.events.ObjectEvent
import com.n9mtq4.logwindow.listener.ObjectListener
import net.minecraft.launcher.ui.tabs.LauncherTabPanel
import javax.swing.SwingUtilities

/**
 * Created by will on 11/5/15 at 5:23 PM.
 *
 * @author Will "n9Mtq4" Bresnahan
 */
public final class InitWorldManagerTab : ObjectListener {
	
	override fun objectReceived(objectEvent: ObjectEvent, baseConsole: BaseConsole) {
		
		if (!objectEvent.message.equals("tabsafe")) return
		if (objectEvent.contained !is LauncherTabPanel) return
		
		SwingUtilities.invokeLater { 
			val worldManagerTab = WorldManagerTab(objectEvent.contained as LauncherTabPanel, objectEvent.initiatingBaseConsole)
			objectEvent.initiatingBaseConsole.push(worldManagerTab.constructInfo(), "addtab")
		}
		
	}
	
}
